app/WebParsing/PostParser.hs

```diff
@@ -56,44 +52,36 @@ parHtmlToLines tags =
             | otherwise = t:ts
 
 -- Helpers
-categoryParser :: [Tag T.Text] -> T.Text -> Maybe T.Text -> [T.Text] -> SqlPersistM ()
-categoryParser tags fullPostName firstCourse listPartitions = do
-    case parsed of
-        Right (post, categories) -> do
+categoryParser :: [Tag T.Text] -> [[T.Text]] -> T.Text -> Maybe T.Text -> SqlPersistM ()
+categoryParser tags requirements fullPostName firstCourse = do
+    let categories = map parseRequirement requirements
+    let parsedPost = P.parse (postInfoParser fullPostName firstCourse) failedString (innerText tags)
+    liftIO (print categories)
+    case parsedPost of
+        Right post -> do
             postExists <- insertUnique post
             case postExists of
                 Just key -> do
                     addPostCategoriesToDatabase key categories
                 Nothing -> return ()
         Left _ -> do
             liftIO $ print failedString
-    where
-        parsed = case listPartitions of
-            [] -> do
-                P.parse (generalCategoryParser fullPostName firstCourse) failedString (innerText tags)
-            partitionResults -> do
-                post <- P.parse (postInfoParser fullPostName firstCourse) failedString (innerText tags)
-                return (post, partitionResults)
 
 addPostCategoriesToDatabase :: PostId -> [T.Text] -> SqlPersistM ()
 addPostCategoriesToDatabase key categories = do
     mapM_ (insert_ . PostCategory key) (filter isCategory categories)
     where
-        isCategory text = T.length text >= 7 && any (flip T.isInfixOf $ text) ["First", "Second", "Third"] --, "suitable", "Core", "Electives"]
+        isCategory text = T.length text >= 7
 
-parseLi :: [Tag T.Text] -> T.Text
-parseLi liPartition =
-    case P.parse parseCategory failedString (innerText liPartition) of
-        Right c -> c
-        Left _ -> ""
-
-getNumberedPartitions :: [Tag T.Text] -> [[Tag T.Text]]
-getNumberedPartitions tags = 
-    let pTags = partitions (isTagOpenName "p") tags
-    in concatMap (splitWhen (isTagOpenName "br")) pTags
+parseRequirement :: [T.Text] -> T.Text
+parseRequirement requirement = do
+    T.intercalate ", " (map parseSingleReq (filter isReq requirement))
```

Comment:
For now I suggest not doing `T.intercalate`, but instead returning a list of texts. Seems like that's a better approximation of the details, at least for now.